core/src/main/java/org/neo4j/graphalgo/core/WeightMap.java

```diff
@@ -2,48 +2,36 @@
 
 import com.carrotsearch.hppc.LongDoubleHashMap;
 import com.carrotsearch.hppc.LongDoubleMap;
+import org.neo4j.graphalgo.api.WeightMapping;
 
 /**
  * single weight cache
  */
-public final class WeightMapping {
+public final class WeightMap implements WeightMapping {
 
     private final int capacity;
     private LongDoubleMap weights;
 
-    public WeightMapping(final int capacity) {
+    public WeightMap(final int capacity) {
         this.capacity = capacity;
     }
 
-    public WeightMapping(final int capacity, LongDoubleMap weights) {
+    public WeightMap(final int capacity, LongDoubleMap weights) {
         this.capacity = capacity;
         this.weights = weights;
     }
 
     /**
      * return the weight for id or 0.0 if unknown
      */
+    @Override
     public double get(long id) {
-        if (weights != null) {
-            return weights.get(id);
-        }
-        return 0d;
+        return weights.get(id);
```

Comment:
The HPPC maps return 0 as the default, with #6 we have to change this anyway to return the custom default.